[Website] Update macOS to 0.71

Saadnajmi opened the pull request:
## Description

### Why

Update website to list the latest version of React Native macOS.

Saadnajmi commented:
Not sure if I need to add a new versioned doc for `rnm-getting-started.md` or not. Working on it...

chrisglein commented:
> Not sure if I need to add a new versioned doc for `rnm-getting-started.md` or not. Working on it...

If you edit rnm-getting-started.md but not the versioned copies that will change the copy under `next` (e.g. https://microsoft.github.io/react-native-windows/docs/next/rnm-getting-started) but not any previous versions (e.g. https://microsoft.github.io/react-native-windows/docs/0.67/rnm-getting-started).

Since this comment refers to "what is latest", it seems like all copies of rnm-getting-started should be updated?

- [ ] \react-native-windows-samples\website\versioned_docs\version-0.61
- [ ] \react-native-windows-samples\website\versioned_docs\version-0.62
- [ ] \react-native-windows-samples\website\versioned_docs\version-0.63
- [ ] \react-native-windows-samples\website\versioned_docs\version-0.64
- [ ] \react-native-windows-samples\website\versioned_docs\version-0.66
- [ ] \react-native-windows-samples\website\versioned_docs\version-0.67

Saadnajmi commented:
> > Not sure if I need to add a new versioned doc for `rnm-getting-started.md` or not. Working on it...
> 
> If you edit rnm-getting-started.md but not the versioned copies that will change the copy under `next` (e.g. https://microsoft.github.io/react-native-windows/docs/next/rnm-getting-started) but not any previous versions (e.g. https://microsoft.github.io/react-native-windows/docs/0.67/rnm-getting-started).
> 
> Since this comment refers to "what is latest", it seems like all copies of rnm-getting-started should be updated?
> 
> * [ ]  \react-native-windows-samples\website\versioned_docs\version-0.61
> * [ ]  \react-native-windows-samples\website\versioned_docs\version-0.62
> * [ ]  \react-native-windows-samples\website\versioned_docs\version-0.63
> * [ ]  \react-native-windows-samples\website\versioned_docs\version-0.64
> * [ ]  \react-native-windows-samples\website\versioned_docs\version-0.66
> * [ ]  \react-native-windows-samples\website\versioned_docs\version-0.67

Thanks for the info. I'm pretty sure nothing about getting started has changed per release (It's all just `react-native-macos-init` + `run-macos`).. so I'm tempted to just remove all the versioned versions?

chrisglein commented:
> Thanks for the info. I'm pretty sure nothing about getting started has changed per release (It's all just `react-native-macos-init` + `run-macos`).. so I'm tempted to just remove all the versioned versions?

Yep, you can delete the old ones and remove the need for versioned copies.

Saadnajmi commented:
I removed references of "What the latest version is" from old docs, as well as removing the 0.67 version docs given we never released a 0.67. 
Rather than adding a new 0.71 page, I just updated the "next" version. My understanding is since a 0.71 version doesn't exist, then the "next" version will be used in the 0.71 page?

Saadnajmi commented:
Looks like I don't have write access. Could someone else merge this for me please? :)
